docs/_sources/tutorials/ptq.rst.txt

```diff
@@ -132,11 +135,76 @@ Then all thats required to setup the module for INT8 calibration is to set the f
     auto trt_mod = trtorch::CompileGraph(mod, compile_spec);
 
 If you have an existing Calibrator implementation for TensorRT you may directly set the ``ptq_calibrator`` field with a pointer to your calibrator and it will work as well.
-
 From here not much changes in terms of how to execution works. You are still able to fully use LibTorch as the sole interface for inference. Data should remain
 in FP32 precision when it's passed into `trt_mod.forward`. There exists an example application in the TRTorch demo that takes you from training a VGG16 network on
 CIFAR10 to deploying in INT8 with TRTorch here: https://github.com/NVIDIA/TRTorch/tree/master/cpp/ptq
 
+.. _writing_ptq_python:
+
+How to create your own PTQ application in Python
+----------------------------------------
+
+TRTorch Python API provides an easy and convenient way to use pytorch dataloaders with TensorRT calibrators. ``DataLoaderCalibrator`` class can be used to create
+a TensorRT calibrator by providing desired configuration. The following code demonstrates an example on how to use it
+
+.. code-block:: python
+    self.testing_dataset = torchvision.datasets.CIFAR10(root='./data',
+                                                            train=False,
+                                                            download=True,
+                                                            transform=transforms.Compose([
+                                                                transforms.ToTensor(),
+                                                                transforms.Normalize((0.4914, 0.4822, 0.4465),
+                                                                                     (0.2023, 0.1994, 0.2010))
+                                                            ]))
+
+    self.testing_dataloader = torch.utils.data.DataLoader(self.testing_dataset,
+                                                          batch_size=1,
+                                                          shuffle=False,
+                                                          num_workers=1)
+    self.calibrator = trtorch.ptq.DataLoaderCalibrator(self.testing_dataloader,
+                                                       cache_file='./calibration.cache',
+                                                       use_cache=False,
+                                                       algo_type=trtorch.ptq.CalibrationAlgo.ENTROPY_CALIBRATION_2,
+                                                       device=torch.device('cuda:0'))
+
+    compile_spec = {
+             "input_shapes": [[1, 3, 32, 32]],
+             "op_precision": torch.int8,
+             "calibrator": self.calibrator(),
```

Comment:
This calling syntax is going to trip people up since people will probably assume that the calibrator is an object. Is there a way to do what we need to do without this syntax?

Reply:
`__new__` might be a way to do this

Reply:
Fixed it.